identity-core/src/crypto/signature/ed25519.rs

```diff
@@ -23,7 +23,11 @@ where
 {
   type Private = T;
   type Output = [u8; SIGNATURE_LENGTH];
-
+  /// signs a message.
+  ///
+  /// The private key is expected to be exactly 32 bytes. Other implementations have different conventions for private
+  /// keys, but we follow RFC 8032. What we and the aforementioned RFC call a private key, is called a *secret key* in the paper [High-speed high-security signatures](http://ed25519.cr.yp.to/ed25519-20110926.pdf).
+  /// The following [blog post](https://blog.mozilla.org/warner/2011/11/29/ed25519-keys/) renames the aforementioned private key/secret key to be called a *seed*.
```

Comment:
Nit: clarify that this only generates a signature.

Does the original [High-speed high-security signatures](http://ed25519.cr.yp.to/ed25519-20110926.pdf) paper add anything extra or just confirm where the RFC defintion is from? Can we remove it or should we link it everywhere else too? I found the secret key distinction confusing here, but maybe that's just me.

Aside: we moved away from "secret key" in #409 as it was pointed out by the community to be inaccurate for asymmetric keys.

```suggestion
  /// Computes an EdDSA/Ed25519 signature.
  ///
  ///  The private key must be a 32-byte seed in compliance with [RFC 8032](https://datatracker.ietf.org/doc/html/rfc8032#section-3.2).
/// Other implementations often use another format. See [this blog post](https://blog.mozilla.org/warner/2011/11/29/ed25519-keys/) for further explanation.
```

Reply:
> Does the original High-speed high-security signatures paper add anything extra or just confirm where the RFC defintion is from?

Yes, I wanted to point out that even the RFC deviates from the original phrasing, but realise now that the documentation for a function is not the right place to do that so I will remove it.  

In principle I like the phrasing in your suggestion, but I am wondering if "32-byte seed" should be replaced with "cryptographically secure 32-byte seed" or is that already implied/clear?

Reply:
What does "cryptographically secure" mean for the seed in this context though, that it was obtained from cryptographically secure pseudorandom number generator, can we even enforce that?

Do we just want to advertise that the private keys we (or rather the zebra library) generates are theoretically from a cryptographically strong source? On reflection we can't even guarantee that since it's based on the operating system I believe.

Reply:
> but I am wondering if "32-byte seed" should be replaced with "cryptographically secure 32-byte seed" or is that already implied/clear?

I think it's implied just by being a cryptographic private key basically.

Reply:
>What does "cryptographically secure" mean for the seed in this context though, that it was obtained from cryptographically secure pseudorandom number generator, can we even enforce that? 

By cryptographically secure, I believe one usually means fit for cryptographic purposes, which can indeed mean obtained from a cryptographically secure source.  This is indeed something we cannot enforce programmatically (at least not in this context) so it is maybe better not to give anyone the wrong impression of this being checked on our side. 

>Do we just want to advertise that the private keys we (or rather the zebra library) generate are theoretically from a cryptographically strong source? On reflection we can't even guarantee that since it's based on the operating system I believe.

From what I recall the 32 byte seed is actually created by `iota-crypto` before it is passed to the zebra library, and `iota-crypto` does indeed get its input from the operating system.  I don't know if we should say a few words about how the keys are generated, but this is maybe something that can be handled in a separate PR anyway?

If you don't have any objections I will simply apply your suggestion for now.

Reply:
> I don't know if we should say a few words about how the keys are generated, but this is maybe something that can be handled in a separate PR anyway?

Most of these comments are hidden deep in the code. I don't think a comment is necessary but if we were to add one on what key generation we use I would say somewhere like `KeyPair::new` is more visible.